src/main/java/org/opensearch/knn/index/query/parser/MethodParametersParser.java

```diff
@@ -61,7 +62,7 @@ public static ValidationException validateMethodParameters(final Map<String, ?>
     }
 
     // deserialize for node to node communication
-    public static Map<String, ?> streamInput(StreamInput in) throws IOException {
+    public static Map<String, ?> streamInput(StreamInput in, Function<String, Boolean> minClusterVersionCheck) throws IOException {
```

Comment:
Is this parameter used anywhere?  I think it might be supposed to replace the check on line 72

Reply:
Err, yes I missed replacing that. Nice catch!